unified-scheduler-pool/src/lib.rs

```diff
@@ -2905,8 +2907,10 @@ mod tests {
         let new_scheduler_id = new_scheduler.id();
         Box::new(old_scheduler.into_inner().1).return_to_pool();
 
-        // sleepless_testing can't be used; wait a bit here to see real progress of wall time...
-        sleep(SHORTENED_MAX_POOLING_DURATION * 10);
+        // Wait for old_scheduler to be considered idle by the cleaner (> 300ms old).
+        // This ensures when the cleaner runs, old_scheduler will definitely be idle.
+        sleep(Duration::from_millis(350));
```

Comment:
It would be better to explicitly tie this selection to the `test_max_pooling_duration` value

Reply:
I know 50ms seems like an eternity, but we've seen these types of thresholds still be flaky in CI testing, especially for non-serial testing..